### PR TITLE
[stable/mongodb-replicaset] liveness/startupProbe tuning

### PR DESCRIPTION
**Pull request by wonko**
#### What this PR does / why we need it:

- Adds a startupProbe to the chart, so a larger mongo install won't be terminated by the livenessProbe (will only apply on semverMatch >= 1.16).
- Updated the metrics container to also use the livenessProbe values instead of hardcoded ones.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/mongodb-replicaset/templates/mongodb-statefulset.yaml [hunk 1]
```diff
@@ -221,6 +221,24 @@ spec:
             failureThreshold: {{ .Values.livenessProbe.failureThreshold }}
             periodSeconds: {{ .Values.livenessProbe.periodSeconds }}
             successThreshold: {{ .Values.livenessProbe.successThreshold }}
+{{- if semverCompare ">=1.16-0" .Capabilities.KubeVersion.GitVersion -}}
```
Comment (steven-sheehy) [inline stable/mongodb-replicaset/templates/mongodb-statefulset.yaml]: Looks like startupProbe is not enabled by default (e.g. marked beta) until 1.18. Would prefer this be set to that.

File: stable/mongodb-replicaset/values.yaml [hunk 2]
```diff
@@ -198,3 +198,12 @@ livenessProbe:
   failureThreshold: 3
   periodSeconds: 10
   successThreshold: 1
+
+# Startup probe, will only be added since Kubernetes v1.16 or higher
+# When you have a lot of collections/indices, MongoDB need some time to start
+startupProbe:
+  initialDelaySeconds: 5
```
Comment (steven-sheehy) [inline stable/mongodb-replicaset/values.yaml:206]: These don't match the README. Please sync them

File: stable/mongodb-replicaset/templates/mongodb-statefulset.yaml [hunk 3]
```diff
@@ -296,8 +314,26 @@ spec:
...
                   --test
-            initialDelaySeconds: 30
-            periodSeconds: 10
+            initialDelaySeconds: {{ .Values.livenessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.livenessProbe.periodSeconds }}
+{{- if semverCompare ">=1.16-0" .Capabilities.KubeVersion.GitVersion -}}
+          startupProbe:
+            exec:
+              command:
+                - mongo
```
Comment (steven-sheehy) [inline stable/mongodb-replicaset/templates/mongodb-statefulset.yaml]: This shouldn't test mongo directly but run the same test as the readiness probe. The readiness probe already tests the mongo connection itself.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]
/assign
/ok-to-test

**Comment by wonko** [comment]
Addressed all remarks @steven-sheehy

**Comment by steven-sheehy** [comment]
Please fix conflicts and ensure all tests pass

**Review by steven-sheehy — APPROVED** [review]
/lgtm

**Pull request merged**